### ISSUE TITLE
iOS Build Fails Due to AESCrypt-Related Errors

### ISSUE THREAD

**Issue by DorianMazur** [Tencent/MMKV#1392]
## Note:

New changes in AESCrypt.cpp and AESCrypt.h are causing iOS build to fail
I am talking about this commit -> https://github.com/Tencent/MMKV/commit/b154a2387605646fcea07b1a2f36cbe8a6fc185a

Many errors related to AESCrypt, such as this when trying to compile.
Related react-native-mmkv issue -> https://github.com/mrousavy/react-native-mmkv/issues/723

### The version of MMKV

react-native-mmkv is using git submodule, pointing to this commit
https://github.com/Tencent/MMKV/commit/b154a2387605646fcea07b1a2f36cbe8a6fc185a

### The platform of MMKV

IOS

### The installation of MMKV

Cocoapods

### What's the log of MMKV when that happened?
```
❌ (node_modules/react-native-mmkv/MMKV/Core/aes/AESCrypt.cpp:192:16)

190 |
191 | // check if AESCrypt is encrypt-decrypt full-duplex

192 | void AESCrypt::testAESCrypt() {
| ^ cannot define or redeclare 'testAESCrypt' here because namespace 'mmkv' does not enclose namespace 'AESCrypt'
193 | testRandomPlaceHolder();
194 |
195 | const uint8_t plainText[] = "Hello, OpenSSL-mmkv::AESCrypt::testAESCrypt() with AES CFB 128.";
```

**Comment by mrousavy** [comment]
> Cocoapods

react-native-mmkv uses MMKV as a submodule, not a cocoapod

**Comment by lingol** [comment]
It's fixed in the dev branch. 
This error only happens on a debug build. You should have no problem with a **release build** for the time being.

**Comment by mrousavy** [comment]
Thanks @lingol! 🙏

**Comment by lingol** [comment]
[v2.0.0](https://github.com/Tencent/MMKV/releases/tag/v2.0.0) is out.

**Comment by TechWilk** [comment]
We've just run into the same version and haven't yet migrated to the 2.0.x releases yet. Is there any way this fix can be backported to the 1.3.x branch?

**Comment by lingol** [comment]
I will cherry pick it back to 1.3.x branch if I haven't done so. However v1.3.12 has just been released so next version won't come soon.
